src/emqx_broker.erl

```diff
@@ -434,10 +434,11 @@ pick(Topic) ->
 
 init([Pool, Id]) ->
     true = gproc_pool:connect_worker(Pool, {Pool, Id}),
-    {ok, #{pool => Pool, id => Id}}.
+    {Worker, WorkerMRef} = emqx_router:start_trans_worker(),
```

Comment:
spawn_link to make it simple.
we need a test case to ensure that emqx_broker processes will never `trap_exit` in the future.

Reply:
spawn_link is bidirectional which is not ideal at this point.